java/src/main/java/org/digma/intellij/plugin/idea/psi/java/JavaEndpointNavigationProvider.java

```diff
@@ -97,6 +101,33 @@ private void buildEndpointAnnotations(@NotNull SearchScope searchScope) {
         });
     }
 
+    @Nullable
+    protected PsiFile getPsiFile(@NotNull VirtualFile virtualFile) {
+        final PsiFile psiFile;
+        try {
+            psiFile = PsiUtils.uriToPsiFile(virtualFile.getUrl(), project);
+        } catch (PsiFileNotFountException e) {
+            // very unlikely
+            return null;
+        }
+        return psiFile;
+    }
```

Comment:
That is not the way to get PsiFile from VirtualFile.
the way is:
PsiFile psiFile = PsiManager.getInstance(project).findFile(virtualFile);

PsiUtils.uriToPsiFile is for very specific use cases when we really only have a url to a file